File: Task/Create-an-HTML-table/MATLAB/create-an-html-table.m
function htmltable(fid,table,Label)
   fprintf(fid,'<table>\n  <thead align = "right">\n');
   if nargin<3,
       fprintf(fid,'    <tr><th></th><td>X</td><td>Y</td><td>Z</td></tr>\n  </thead>\n  <tbody align = "right">\n');
   else
       fprintf(fid,'    <tr><th></th>');
       fprintf(fid,'<td>%s</td>',Label{:});
       fprintf(fid,'</tr>\n  </thead>\n  <tbody align = "right">\n');
   end;
   fprintf(fid,'    <tr><td>%2i</td><td>%5i</td><td>%5i</td><td>%5i</td></tr>\n', [1:size(table,1);table']);
   fprintf(fid,'  </tbody>\n</table>\n');
end

File: Task/Create-an-HTML-table/R/create-an-html-table.r
html_table <- function(headings, nrows){
  ncols <- length(headings)
  pad_table <- function(s, char2="d"){
    pad <- paste0("<t", char2, ">", s, "</t", char2, ">", collapse="")
    paste0("<tr>", pad, "</tr>", collapse="")
  }
  theader <- pad_table(c("", headings), char2="h")
  gen_row <- function(n) c(n, sample(9999, ncols))
  tbody <- lapply(1:nrows, gen_row) |> sapply(pad_table)
  cat("<table>", theader, sep="\n")
  writeLines(tbody)
  cat("</table>\n")
}

html_table(c("X", "Y", "Z"), 3)
